flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -80,8 +80,8 @@ def AnyRefOfConstantSizeAggregateType : TypeConstraint<
 // Memory SSA operations
 //===----------------------------------------------------------------------===//
 
-def fir_AllocaOp : fir_Op<"alloca", [AttrSizedOperandSegments,
-    MemoryEffects<[MemAlloc<AutomaticAllocationScopeResource>]>]> {
+def fir_AllocaOp : fir_Op<"alloca",
+    [AttrSizedOperandSegments, DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]> {
```

Comment:
Just FYI: if you can remove code in `fir::AllocaOp::getEffects` as I suggest below, then you can declare the `MemAlloc` effect in the operation definition like this: https://github.com/llvm/llvm-project/blob/32ebf635c2be171b01a288b00b3e64b8de72e61a/mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td#L249

This way you won't need to change the cpp files.